src/sqlfluff/core/rules/std/L025.py

```diff
@@ -53,7 +53,7 @@ def _lint_references_and_aliases(
         for r in references:
             tbl_ref = r.extract_reference(level=2)
             if tbl_ref:
-                tbl_refs.add(tbl_ref[0])
+                tbl_refs.add(tbl_ref.part)
```

Comment:
Tidying up the code, only _slightly_ related to the PR. Here `.part` is referring to a field of `ObjectReferencePart`, a `NamedTuple` whose structure changes slightly in this PR (`segment` becomes `segments`).